pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -267,10 +272,30 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		})
 	})
 	registerHandlers(c.pods.informer, c.queue, "Pods", c.pods.onEvent, nil)
-
+	if err := updateSPIFFERegistry(kubeClient, c.clusterID); err != nil {
+		log.Errorf("Failed to update the trust domain for cluster %v: %v", c.clusterID, err)
+	}
 	return c
 }
 
+func updateSPIFFERegistry(kubeClient kubelib.Client, clusterID string) error {
+	cm, err := kubeClient.CoreV1().ConfigMaps(IstioNamespace).Get(context.TODO(), constants.IstioMeshConfigConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to read config map for trust domain: %v", err)
+	}
+	if cm == nil || cm.Data == nil {
+		log.Debugf("MeshConfig empty, skip trust domain updates")
+		return nil
+	}
+	v := cm.Data[constants.IstioMeshConfigEntryName]
+	mesh := meshconfig.MeshConfig{}
+	if err := gogoprotomarshal.ApplyYAML(v, &mesh); err != nil {
```

Comment:
to be safe, use `ApplyMeshConfigDefaults` instead

Reply:
done.